File: Python/ExampleTests/production/RecordFilter.py
from .IDForm import IDForm
from .IDValue import IDValue


class RecordFilter:
    """Holds IDValue records and sums the ones carrying a given ID."""

    def __init__(self):
        self._entries: list[IDValue] = []

    def add(self, entry: IDValue) -> None:
        self._entries.append(entry)

    def sum_by_label(self, filter_label: IDForm) -> int:
        return sum(e.value for e in self._entries if e.id == filter_label)

File: Rust/ExampleTests/production/recordfilter.rs
use super::*;

/// Holds IDValue records and sums the ones carrying a given ID.
#[derive(Debug, Clone, Default)]
pub struct RecordFilter {
    entries: Vec<IDValue>,
}

impl RecordFilter {
    pub fn new() -> Self { Self::default() }

    pub fn add(&mut self, entry: IDValue) { self.entries.push(entry); }

    pub fn sum_by_label(&self, filter_label: &IDForm) -> i32 {
        self.entries.iter()
            .filter(|e| &e.id == filter_label)
            .map(|e| e.value)
            .sum()
    }
}
